test/with_dummyserver/test_socketlevel.py

```diff
@@ -911,6 +911,54 @@ def echo_socket_handler(listener):
         r = conn.urlopen('GET', url, retries=0)
         self.assertEqual(r.status, 200)
 
+    def test_connect_ipv6_addr(self):
+        ipv6_addr = '2001:4998:c:a06::2:4008'
+
+        def echo_socket_handler(listener):
+            sock = listener.accept()[0]
+
+            buf = b''
+            while not buf.endswith(b'\r\n\r\n'):
+                buf += sock.recv(65536)
+            s = buf.decode('utf-8')
+            if s.startswith('CONNECT %s:443' % (ipv6_addr,)):
+                sock.send(b'HTTP/1.1 400 Bad Request\r\n\r\n')
+                sock.close()
+                return
```

Comment:
Let's not add this block: instead, just have an `elif` block at the bottom of the next `if` block that forcibly-closes the socket. That'll cause the client to error out anyway.